### ISSUE TITLE
Track number metadata not included

### ISSUE THREAD

**Issue by Batwam** [NickvisionApps/Parabolic#743]
Hello,

I just downloaded a playlist including the track numbers. This works for the file name but I noticed that the track number is not included in the metadata. I initially thought to suggest doing it but I can see from #435 that this has been implemented. Is this meant to work? if so, it didn't work for me.

Ubuntu 23.10
Parabolic v2023.10.2 (flatpak)

**Comment by Batwam** [comment]
ok, I have noted that the track numbers are now showing when I download songs. I was going to mark this as closed but couldn't quite let it go as I couldn't understand why this changed. I then remembered that the only thing I changed was to turn on the option to scrub the Source Metadata. So, I went back, turn that back off and the track numbers disappeared again. bingo!

`exiftool` data with Metadata scrubbing on (left) and off (right):
![Screenshot from 2024-02-25 22-05-43](https://github.com/NickvisionApps/Parabolic/assets/56710655/8bdbb44c-6ce4-43a5-9f94-eb59c6a74fbb)

as you can see, the right hand side has the video link but it's also missing the track number (the rest of the data cropped at the top is identical between both files). Even the order between Artist/Year/Album is different between both sides so,it looks like this option does more than just removing the comment, it also adds the track number?

**Comment by nlogozzo** [comment]
@Batwam Could you test this release and let me know what happens: https://github.com/NickvisionApps/Parabolic/releases/tag/2024.9.0-beta1

**Comment by Batwam** [comment]
just tried installing the snap version. it crashes with the following error
`Content snap command-chain for /snap/tube-converter/576/gpu-2404/bin/gpu-2404-provider-wrapper not found: ensure slot is connected`

**Comment by Batwam** [comment]
More of t a beta version issue but I can't test, flatpak starts but I get the following error when inputing the url:
![image](https://github.com/user-attachments/assets/ffc0dd5a-4bfd-404a-83b7-6f3fe71a7052)

it works fine when using yt-dlp directly though.

**Comment by nlogozzo** [comment]
> just tried installing the snap version. it crashes with the following error
> `Content snap command-chain for /snap/tube-converter/576/gpu-2404/bin/gpu-2404-provider-wrapper not found: ensure slot is connected`

@soumyaDghosh

**Comment by nlogozzo** [comment]
> More of t a beta version issue but I can't test, flatpak starts but I get the following error when inputing the url (I have selected my browser in the preferences even though this isn't something I had to do in the previous versions):
> ![image](https://github.com/user-attachments/assets/ffc0dd5a-4bfd-404a-83b7-6f3fe71a7052)
> 
> it works fine when using yt-dlp directly though.

What's the link and what's the browser?

**Comment by soumyaDghosh** [comment]
> > just tried installing the snap version. it crashes with the following error
> > `Content snap command-chain for /snap/tube-converter/576/gpu-2404/bin/gpu-2404-provider-wrapper not found: ensure slot is connected`
> 
> @soumyaDghosh 

I have no idea, why your snaps are facing this! I guess it's time to get serious about this.

**Comment by Batwam** [comment]
If that helps, I had  a look at the content of the folder and there is no gpu-2404 folder in it. There is also no file called *provider-wrapper* anywhere within `/snap/tube-converter`.

**Comment by Batwam** [comment]
> > More of t a beta version issue but I can't test, flatpak starts but I get the following error when inputing the url (I have selected my browser in the preferences even though this isn't something I had to do in the previous versions):
> > ![image](https://private-user-images.githubusercontent.com/56710655/365420885-ffc0dd5a-4bfd-404a-83b7-6f3fe71a7052.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3OTQ0NjcsIm5iZiI6MTcyNTc5NDE2NywicGF0aCI6Ii81NjcxMDY1NS8zNjU0MjA4ODUtZmZjMGRkNWEtNGJmZC00MDRhLTgzYjctNmYzZmU3MWE3MDUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA4VDExMTYwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyZWEwNmE1N2YzNGRjNmFkNWFiMDk4NjQ3NmM2OGE1ZTFhNmM4NThmYTI3NzgxZjNiNWVhNDg4N2VlM2U5YjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8LxPeWQ4GLvTXvWZgW-lziyN3-KcnWcMOv_mu1-n_bM)
> > it works fine when using yt-dlp directly though.
> 
> What's the link and what's the browser?

it's the same with any song. For example this Creative Common song: https://youtu.be/5yIbZVOv438?si=RDyV4tLhapccgSGS
browser is firefox. Same thing is I specify chrome (which is also installed). I haven't specified a link to a Cookies files as I never had to before and don't need to do it with yt-dlp. not sure if this is required or not...

**Comment by soumyaDghosh** [comment]
> If that helps, I had  a look at the content of the folder and there is no gpu-2404 folder in it. There is also no file called *provider-wrapper* anywhere within `/snap/tube-converter`.

I know why it's happening, and how to fix it. I'm just furious why is it happening. It should not be the case!

Anyways, thanks a lot for taking your time to look into it. To fix it run


```
sudo snap connect tube-converter:gpu-2404 mesa-2404
```

**Comment by Batwam** [comment]
I ran the command above, now I'm getting this:
`Content snap command-chain for /snap/tube-converter/576/gnome-platform/command-chain/desktop-launch not found: ensure slot is connected`

**Comment by soumyaDghosh** [comment]
```
sudo snap connect tube-converter:gnome-46-2404 gnome-46-2404
sudo snap connect tube-converter:ffmpeg-2404 ffmpeg-2404
```

**Comment by Batwam** [comment]
yeah, it's starting now, it does show the following errors in the terminal though:
```
libpxbackend-1.0.so: cannot open shared object file: No such file or directory
Failed to load module: /home/anthony/snap/tube-converter/common/.cache/gio-modules/libgiolibproxy.so
```
as well as the following Warnings:
```
(org.nickvision.tubeconverter.gnome:919141): Adwaita-WARNING **: 08:01:03.832: AdwNavigationPage 0x60fd514ef250 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.

(org.nickvision.tubeconverter.gnome:919141): Adwaita-WARNING **: 08:01:03.833: AdwNavigationPage 0x60fd5153c110 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.
```

interestingly though, downloading the link above does work with the snap version (didn't even have to specify a browser, in fact, the option isn't even there... Perhaps a permission/access issue with the flatpak version then?

**Comment by nlogozzo** [comment]
> yeah, it's starting now, it does show the following errors in the terminal though:
> ```
> libpxbackend-1.0.so: cannot open shared object file: No such file or directory
> Failed to load module: /home/anthony/snap/tube-converter/common/.cache/gio-modules/libgiolibproxy.so
> ```
> as well as the following Warnings:
> ```
> (org.nickvision.tubeconverter.gnome:919141): Adwaita-WARNING **: 08:01:03.832: AdwNavigationPage 0x60fd514ef250 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.
> 
> (org.nickvision.tubeconverter.gnome:919141): Adwaita-WARNING **: 08:01:03.833: AdwNavigationPage 0x60fd5153c110 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.
> ```
> 
> interestingly though, downloading the link above does work with the snap version (didn't even have to specify a browser, in fact, the option isn't even there... Perhaps a permission/access issue with the flatpak version then?

The flatpak version also is not supposed to show this option...and I can confirm that it doesn't. 

Is there something weird with ur flatpak installation?

**Comment by Batwam** [comment]
For clarity, the messages above come from the snap installation. The snap version now launches and successfully downloads. 

The flatpak version launches but fails to download as per the errors above about the browser.

**Comment by nlogozzo** [comment]
![image](https://github.com/user-attachments/assets/75215258-e34c-43d7-9d98-03e0a888cae4)

It works for me no problem on Flatpak.

**Comment by nlogozzo** [comment]
It does seem tho as the flatpak version cannot correctly detect that it is running under flatpak. But as long as you keep Cookies from Browser to None and don't upload a cookies file, it should work just fine.

I'm going to fix the detection tho and maybe it will fix your flatpak too.

**Comment by Batwam** [comment]
Ok, could it be because firefox is running as a snap and the files aren't where parabolic expects to find them?

The snap version of parabolic is working fine so I'll give that a go for now.

**Comment by nlogozzo** [comment]
> Ok, could it be because firefox is running as a snap and the files aren't where parabolic expects to find them?

The problem is Flatpak sandboxes the filesystem so yt-dlp can't find anything where it expects it to be.

So on flatpak and snap we disable the choice of cookies for the browser and only allow manually uploading a cookies file.

**Comment by Batwam** [comment]
ok, I'm definitely seeing the option for choice of cookies for the browser with the flatpak version (not the snap). Are you saying that it shouldn't be there?
![image](https://github.com/user-attachments/assets/49a9a255-400c-4b55-a38d-3fbc89e2263f)

**Comment by nlogozzo** [comment]
> Are you saying that it shouldn't be there?

Correct. You should leave that as `None` for now. It will disappear in the next beta.

**Comment by nlogozzo** [comment]
@Batwam Either way...is the track number included in the metadata on the snap version? 😅

**Comment by Batwam** [comment]
yeah, I haven't forgotten about the initial issue, just struggling to even get to a point where I can download files and check 😅

I just tried with a new playlist and get the following (using the snap):
`ERROR: Postporcessing: ffprobe and ffmpeg not found. Please install or provide the path using --ffmpeg-location`
these programes are in the usual `\usr\bin` location...

**Comment by soumyaDghosh** [comment]
> ```
> sudo snap connect tube-converter:gnome-46-2404 gnome-46-2404
> sudo snap connect tube-converter:ffmpeg-2404 ffmpeg-2404
> ```
> 
> Can you kindly try this? @Batwam 

@Batwam You didn't connect both of these. Run the second one properly.

**Comment by Batwam** [comment]
I did, several times but I'm still having the same issue

**Comment by nlogozzo** [comment]
> I did, several times but I'm still having the same issue

I will releasing a new beta later tonight. Hopefully that flatpak will work for you.

**Comment by Batwam** [comment]
see connections list: ![image](https://github.com/user-attachments/assets/bf156403-31b5-4616-ad1d-265e8dacb5e8)

**Comment by Batwam** [comment]
just tried the beta2 version in flatpak, still the same issue. I even tried to generate a cookies text file ([using this code](https://pastebin.com/gxKHDQty)) and add it to the preferences an still face the same issue. Frankly, having to extract the cookies from the sqlite database to a text file for this seems overly convoluted. I'd rather use the snap if that allows to avoid this extra manual step.

**Comment by nlogozzo** [comment]
Using cookies is NOT REQUIRED for Parabolic to work. A cookies file is only required when trying to download age-restricted videos, according to yt-dlp. 

In Preferences you can leave `Cookies from Browser` as `None` and clear the `Cookies File` and leave it blank. And your downloads should work no problem.

FYI, there are extensions available for all major browsers that will extract a cookies txt file for you.

**Comment by Batwam** [comment]
ok, that's good to know. I used the python code mentioned above and that worked too.
I'm not planning to download age-restricted content so I set the browser to None and it's now working. 

Back to the original issue, I have run a test with a playlist and found that the track number isn't embedded. Is there a specific option I need to select for this (I have the option to include the Number Title on, although I noticed that it  is off by default and I have to reactivate it every time). 

Also, the option to Remove Source Data doesn't appear to be doing anything as the data is still there in the Exif field "User Defined Text". I compared the output from exiftool with the option on/off and they are identical.

**Comment by nlogozzo** [comment]
> Back to the original issue

Finally found a fix for this

> Also, the option to Remove Source Data doesn't appear to be doing anything as the data is still there in the Exif field "User Defined Text". I compared the output from exiftool with the option on/off and they are identical.

And for this!

**Comment by nlogozzo** [comment]
[V2025.1.3](https://github.com/NickvisionApps/Parabolic/releases/tag/2025.1.3) is here with a fix for this issue!

*Please allow some time for it to be available on Flathub*
